client/src/components/Upload.vue

```diff
@@ -49,6 +49,31 @@ export default {
         return `${formatSize(totalProgress)} of ${formatSize(totalSize)}`;
       }
     },
+    /**
+     * Calculates the field properties for each pending upload
+     * @param {('label'|'disabled'|'default')} field - the field property we want the current state for
+     * @param {{createFolder: boolean, files: Array}} pendingUpload
+     */
+    getFilenameInputState(field, pendingUpload) {
+      if (field === "label") {
+        let label = pendingUpload.createFolder ? "Folder" : "File";
+        label += " Name";
+        if (!pendingUpload.createFolder && pendingUpload.files.length > 1) {
+          label += "s";
+        }
+        return label;
+      } else if (field === "disabled") {
+        return (
+          pendingUpload.uploading ||
+          (!pendingUpload.createFolder && pendingUpload.files.length > 1)
+        );
+      } else if (field === "hint") {
+        return !pendingUpload.createFolder && pendingUpload.files.length > 1
+          ? "default filenames are used"
+          : "";
+      }
+      return false;
+    },
```

Comment:
I don't generally like the practice of actively returning different types from a function, I think it'd be better to split this up into its 3 respective methods, and would also simplify the syntax. Lmk what you think.

Reply:
That's a good point.  I think that if we start adding any more functionality to the 'PendingUpload' object in Upload.vue that it should probably be broken down into it's own Vue component.  It's getting close (at least for my preference) to cluttering up the Upload.vue component.